js/ui/scheduler/appointmentSettingsGenerator.js

```diff
@@ -184,38 +188,36 @@ export default class AppointmentSettingsGenerator {
         });
     }
 
-    _updateGroupIndices(appointments, itemResources) {
-        const workspace = this.scheduler.getWorkSpace();
-
-        if(workspace.isVirtualScrolling()) {
-            const groupIndices = workspace._isVerticalGroupedWorkSpace()
-                ? workspace._getGroupIndexes(itemResources)
-                : [0];
+    _createExtremeRecurrenceDates(rawAppointment, groupIndex) {
+        const dateRange = this._getGroupDateRange(groupIndex);
 
-            groupIndices.forEach(groupIndex => {
-                appointments.forEach(appointment => appointment.groupIndex = groupIndex);
-            });
-        }
-    }
-
-    _createExtremeRecurrenceDates(rawAppointment) {
-        const dateRange = this.scheduler._workSpace.getDateRange();
-        const startViewDate = this.scheduler.appointmentTakesAllDay(rawAppointment) ? dateUtils.trimTime(new Date(dateRange[0])) : dateRange[0];
+        const startViewDate = this.scheduler.appointmentTakesAllDay(rawAppointment)
+            ? dateUtils.trimTime(new Date(dateRange[0]))
```

Comment:
Should we wrap dateRange[0] with new Date() in this case?

Reply:
Copy-Paste. I'll correct this.